### PR TITLE
Fixes #23603: Add rpkg unpacking in the rust port

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23603

### CODE REVIEW

File: relay/sources/rudder-package/src/rpkg/webapp_xml.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        };
+        let mut updated_xml = self.elements.clone();
+        updated_xml
+            .children
+            .push(xmltree::XMLNode::Element(plugin_element));
+        let _ = updated_xml.write_with_config(
+            File::create(self.path.clone())?,
+            self.emitter_config.clone(),
+        );
+        Ok(())
```
Comment (amousset) [inline relay/sources/rudder-package/src/rpkg/webapp_xml.rs:70]: don't we need to also update self.elements?

File: relay/sources/rudder-package/tests/archive/expected_dir_content/share/python/glpi.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import requests, json
```
Comment (amousset) [inline relay/sources/rudder-package/tests/archive/expected_dir_content/share/python/glpi.py:1]: you could create a test plugin with placeholder scripts instead

**Review by amousset — APPROVED** [review]

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
PR rebased

**Pull request merged**